api/controllers/WebActionsController.js

```diff
@@ -8,6 +8,7 @@ const Promise = require('bluebird');
 const request = Promise.promisifyAll(require('request'));
 const Mixpanel = require('mixpanel');
 const crypto = require('crypto');
+const sns = require('../services/sns');
```

Comment:
SailsJS handles services in a special way so you don't have to `require` it.

https://0.12.sailsjs.com/documentation/concepts/services

I'd recommend renaming the services/sns.js file you have to services/SnsService.js to be consistent with what's expected of Sails services. And then when you need to call the publishEvent function down below, you can simply do `SnsService.publishEvent`.